### ISSUE TITLE
bug: TypeError: Cannot read property 'type' of undefined\nOccurred while linting

### ISSUE THREAD

**Issue by aladdin-add** [eslint/eslint#11521]
it was detected by fuzzer. see https://travis-ci.org/eslint/eslint/jobs/507427588

**Tell us about your environment**

* **ESLint Version:** master
* **Node Version:** n/w
* **npm Version:** n/w

**What parser (default, Babel-ESLint, etc.) are you using?**
default
**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js

            "config": {

                "rules": {

                    "implicit-arrow-linebreak": [

                        2,

                        "beside"

                    ]

                },

                "parserOptions": {

                    "sourceType": "script",

                    "ecmaVersion": 2018

                }

            },
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js

```

```bash

```

**What did you expect to happen?**
TBF

**What actually happened? Please include the actual, raw output from ESLint.**
TBF

**Are you willing to submit a pull request to fix this bug?**
not atm :(

**Comment by aladdin-add** [comment]
it might be introduced in https://github.com/eslint/eslint/commit/29dbca73d762a809adb2f457b527e144426d54a7

**Comment by not-an-aardvark** [comment]
Smaller reproducible example (causes a crash with `implicit-arrow-linebreak` on the default settings):

```js
() => /*
*/ (a = () => b) => c
```

(I've been working on an enhancement to the fuzzer that minimizes the examples that it produces, but the tools isn't ready yet.)

(Demo: [here](https://eslint.org/demo/#eyJ0ZXh0IjoiLyogZXNsaW50IGltcGxpY2l0LWFycm93LWxpbmVicmVhazogb2ZmICovXG5cbi8vIENoYW5naW5nIFwib2ZmXCIgdG8gXCJlcnJvclwiIGluIHRoZSBsaW5lIGFib3ZlIGNhdXNlcyB0aGUgZGVtbyB0byBjcmFzaC5cblxuKCkgPT4gLypcbiovIChhID0gKCkgPT4ge30pID0+IGMiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjYsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=))
